src/meshlink.c

```diff
@@ -1105,68 +825,120 @@ static struct timeval idle(event_loop_t *loop, void *data) {
 
 // Get our local address(es) by simulating connecting to an Internet host.
 static void add_local_addresses(meshlink_handle_t *mesh) {
-	char host[NI_MAXHOST];
-	char entry[MAX_STRING_SIZE];
+	struct sockaddr_storage sn;
+	socklen_t sl = sizeof(sn);
 
 	// IPv4 example.org
 
-	if(getlocaladdrname("93.184.216.34", host, sizeof(host), mesh->netns)) {
-		snprintf(entry, sizeof(entry), "%s %s", host, mesh->myport);
-		append_config_file(mesh, mesh->name, "Address", entry);
+	if(getlocaladdr("93.184.216.34", (struct sockaddr *)&sn, &sl, mesh->netns)) {
+		((struct sockaddr_in *)&sn)->sin_port = ntohs(atoi(mesh->myport));
+		meshlink_hint_address(mesh, (meshlink_node_t *)mesh->self, (struct sockaddr *)&sn);
 	}
 
 	// IPv6 example.org
 
-	if(getlocaladdrname("2606:2800:220:1:248:1893:25c8:1946", host, sizeof(host), mesh->netns)) {
-		snprintf(entry, sizeof(entry), "%s %s", host, mesh->myport);
-		append_config_file(mesh, mesh->name, "Address", entry);
+	sl = sizeof(sn);
+
+	if(getlocaladdr("2606:2800:220:1:248:1893:25c8:1946", (struct sockaddr *)&sn, &sl, mesh->netns)) {
+		((struct sockaddr_in6 *)&sn)->sin6_port = ntohs(atoi(mesh->myport));
+		meshlink_hint_address(mesh, (meshlink_node_t *)mesh->self, (struct sockaddr *)&sn);
 	}
 }
 
 static bool meshlink_setup(meshlink_handle_t *mesh) {
-	if(mkdir(mesh->confbase, 0777) && errno != EEXIST) {
-		logger(mesh, MESHLINK_DEBUG, "Could not create directory %s: %s\n", mesh->confbase, strerror(errno));
+	if(!config_init(mesh)) {
```

Comment:
Unlike `meshlink_read_conf()`, `meshlink_setup()` never called `main_config_lock()` to lock the `meshlink.conf` file.

Reply:
You're right, it should do that.